include/misc/stack.h

```diff
@@ -13,85 +13,80 @@
 #define ZEPHYR_INCLUDE_MISC_STACK_H_
 
 #include <misc/printk.h>
+#include <logging/log.h>
 
-#if defined(CONFIG_INIT_STACKS)
 static inline size_t stack_unused_space_get(const char *stack, size_t size)
 {
 	size_t unused = 0;
 
-#ifdef CONFIG_STACK_SENTINEL
-	/* First 4 bytes of the stack buffer reserved for the sentinel
-	 * value, it won't be 0xAAAAAAAA for thread stacks.
-	 */
-	const unsigned char *checked_stack = (const unsigned char *)stack + 4;
-#else
-	const unsigned char *checked_stack = (const unsigned char *)stack;
-#endif
+	if (IS_ENABLED(CONFIG_INIT_STACKS)) {
```

Comment:
personally, I prefer to have early exit:
```
if (!IS_ENABLED(CONFIG_INIT_STACKS)) {
    return;
}
```
one less indentation but it may be just my personal preference.